core/src/com/unciv/logic/map/tile/Tile.kt

```diff
@@ -311,6 +311,9 @@ open class Tile : IsPartOfGameInfoSerialization {
     fun changeImprovement(improvementStr: String?) {
         improvementIsPillaged = false
         improvement = improvementStr
+
+        if (resource != null && owningCity != null)
+            owningCity!!.civ.cache.updateCivResources()
```

Comment:
If the responsibility on updating resources is here and not whatever is changing the improvement, then a reminder that you need to update resources if the improvement has provides/consumes a resource

Reply:
It's actually worse than that... because the *previous* improvement could have been providing a resource and now we replaced it...
I think I'll just update resources anyway instead of if/elsing all these conditions, we can always improve runtime later if we see it's mature optimization :)